### ISSUE TITLE
Accessing a member variable before declaring a variable that would overshadow returns null

### ISSUE THREAD

**Issue by isaaccp** [godotengine/godot#74015]
### Godot version

master: 6296b46008fb8d8e5cb9b60af05fa1ea26b8f600

### System information

Windows 11

### Issue description

Accessing a class member variable in before a local variable declaration that would overshadow it makes the class member variable access return null.

I'd expect either:
- to get an error for accessing "damage" before it was declared
- to get the value of the member

See code example below.

### Steps to reproduce

```
@extends Node2D

var damage = 0

func _ready():
  if damage == 0:
    print("attribute is 0: ", damage)
  else:
    print("attribute is non-zero: ", damage)
  var damage = 1
  print("local is: ", damage)
```
outputs:

attribute is non-zero: `<null>`
local is: 1

![image](https://user-images.githubusercontent.com/8712/221455885-feba52e2-c956-4c36-9a03-b00cda3c86be.png)

### Minimal reproduction project

N/A

**Comment by akien-mga** [comment]
As seen in #74459, this can also trigger a crash when using type hints:
```gdscript
var my_array = []

func _ready() -> void:
	my_array.clear()
	
	var my_array := []
```

Crashes with:
```
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.dev.custom_build (d81e6ee024a8c64b80ac25c96b33c749ba1db79d)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x36940) [0x7ff76ef3b940] (??:0)
[2] Array::clear() (/home/akien/Projects/godot/godot.git/core/variant/array.cpp:109)
[3] void call_with_validated_variant_args_helper<Array>(Array*, void (Array::*)(), Variant const**, IndexSequence<>) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:355 (discriminator 4))
[4] void call_with_validated_variant_args<Array>(Variant*, void (Array::*)(), Variant const**) (/home/akien/Projects/godot/godot.git/./core/variant/binder_common.h:603)
[5] /home/akien/Projects/godot/godot.git/bin/godot.linuxbsd.editor.dev.x86_64() [0xa2a1117] (/home/akien/Projects/godot/godot.git/core/variant/variant_call.cpp:124)
[6] /home/akien/Projects/godot/godot.git/bin/godot.linuxbsd.editor.dev.x86_64() [0xa1eab8b] (/home/akien/Projects/godot/godot.git/core/variant/variant_call.cpp:2190)
[7] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript_vm.cpp:2124)
[8] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript.cpp:1826)
[9] bool Node::_gdvirtual__ready_call<false>() (/home/akien/Projects/godot/godot.git/scene/main/node.h:242 (discriminator 1))
[10] Node::_notification(int) (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:156)
[11] Node::_notificationv(int, bool) (/home/akien/Projects/godot/godot.git/./scene/main/node.h:46 (discriminator 14))
[12] CanvasItem::_notificationv(int, bool) (/home/akien/Projects/godot/godot.git/./scene/main/canvas_item.h:45 (discriminator 3))
[13] Node2D::_notificationv(int, bool) (/home/akien/Projects/godot/godot.git/./scene/2d/node_2d.h:37 (discriminator 3))
[14] Object::notification(int, bool) (/home/akien/Projects/godot/godot.git/core/object/object.cpp:792)
[15] Node::_propagate_ready() (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:189)
[16] Node::_propagate_ready() (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:179 (discriminator 3))
[17] Node::_set_tree(SceneTree*) (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:2608)
[18] SceneTree::initialize() (/home/akien/Projects/godot/godot.git/scene/main/scene_tree.cpp:411)
[19] OS_LinuxBSD::run() (/home/akien/Projects/godot/godot.git/platform/linuxbsd/os_linuxbsd.cpp:876)
[20] /home/akien/Projects/godot/godot.git/bin/godot.linuxbsd.editor.dev.x86_64(main+0x15a) [0x52dcd80] (/home/akien/Projects/godot/godot.git/platform/linuxbsd/godot_linuxbsd.cpp:75)
[21] /lib64/libc.so.6(+0x23677) [0x7ff76ef28677] (??:0)
[22] /lib64/libc.so.6(__libc_start_main+0x85) [0x7ff76ef28735] (??:0)
[23] /home/akien/Projects/godot/godot.git/bin/godot.linuxbsd.editor.dev.x86_64(_start+0x21) [0x52dcb61] (??:?)
-- END OF BACKTRACE --
```

**Comment by dalexeev** [comment]
Duplicate of #54944. May be fixed by #73427.
